bluebrain/version-bumping/src/bumper.py

```diff
@@ -0,0 +1,305 @@
+            raise ValueError(
+                f"Could not determine versions for {package}. "
+                "Only tag-based versions are supported."
+            )
+
+        return version.parse(max_version.string)
+
+    def build_git_url(self, package_git_url):
+        """
+        Given the package's git URL, build one we can use
+        """
+        if "github.com" in package_git_url:
+            package_git_url = f"https://{package_git_url}"
+        else:
+            if "CI_JOB_TOKEN" in os.environ:
+                logger.debug("CI_JOB_TOKEN found - running on gitlab")
+                package_git_url = (
+                    f"https://gitlab-ci-token:{os.environ['CI_JOB_TOKEN']}@{package_git_url}"
+                )
+            else:
+                logger.debug("CI_JOB_TOKEN not found - SSH should be configured on your machine")
+                package_git_url = f"ssh://{package_git_url}"
+
+        return package_git_url
+
+    def get_latest_source_version(self, package):
+        """
+        Get the latest version of a package in source control
+        """
+        logger.info(f"===> Getting latest source control version for {package}")
+        spack_package = self.get_spack_package(package)
+        try:
+            git_line = spack_package.git
+            print(git_line)
+        except AttributeError:
+            raise ValueError(f"Could not find git source for {package}")
+        git_repo = re.match(GIT_REGEX, git_line).groupdict()["git_url"]
+        logger.debug(f"{package} lives in {git_repo}")
+        git_repo = self.build_git_url(git_repo)
+        proc = subprocess.run(
+            f"git ls-remote --tags {git_repo} | awk -F '/' '{{print $3}}'",
+            shell=True,
+            capture_output=True,
+        )
+        repo_tags = proc.stdout.decode()
+        if "tag_regex" not in self.packages[package]:
+            logger.critical(
+                f"No tag_regex specified for {package}. "
+                f"These are all the tags available: {repo_tags}"
+            )
+            return None, None
+        repo_release_tags = re.findall(self.packages[package]["tag_regex"], repo_tags)
+
+        logger.debug(f"Version tags for {package}: {repo_release_tags}")
+
+        return max(
+            [
+                (
+                    repo_release_tag,
+                    version.parse(
+                        repo_release_tag.replace(self.packages[package].get("tag_strip", ""), "")
+                    ),
+                )
+                for repo_release_tag in repo_release_tags
+            ],
+            key=lambda x: x[1],
+        )
+
+    def add_spack_version(self, package, latest_source_tag, latest_source_version, package_file):
+        logger.info(f"===> Adding version {latest_source_version} for {package}")
+        source_lines = self.get_source_lines(package_file)
+        for idx, line in enumerate(source_lines):
+            if line.strip().startswith("version("):
```

Comment:
could be a regexp... `r"^( )+version\("`. But actually: we don't want to add version directives into nested statements. Those are normally indicative of a programmatic approach to version bumping. Thus a fixed indent of 4 should be checked and used?